dbdimp.c

```diff
@@ -1903,28 +1909,34 @@ MYSQL *mysql_dr_connect(
         }
 #endif
 
+	if ((svp = hv_fetch(hv, "mysql_ssl", 9, FALSE)) && *svp && SvTRUE(*svp))
+          {
 #if defined(DBD_MYSQL_WITH_SSL) && !defined(DBD_MYSQL_EMBEDDED) && \
     (defined(CLIENT_SSL) || (MYSQL_VERSION_ID >= 40000))
-	if ((svp = hv_fetch(hv, "mysql_ssl", 9, FALSE))  &&  *svp)
-        {
-	  if (SvTRUE(*svp))
-          {
 	    char *client_key = NULL;
 	    char *client_cert = NULL;
 	    char *ca_file = NULL;
 	    char *ca_path = NULL;
 	    char *cipher = NULL;
 	    STRLEN lna;
-#if MYSQL_VERSION_ID >= SSL_VERIFY_VERSION && MYSQL_VERSION_ID <= SSL_LAST_VERIFY_VERSION
-            /*
-              New code to utilise MySQLs new feature that verifies that the
-              server's hostname that the client connects to matches that of
-              the certificate
-            */
-	    my_bool ssl_verify_true = 0;
-	    if ((svp = hv_fetch(hv, "mysql_ssl_verify_server_cert", 28, FALSE))  &&  *svp)
-	      ssl_verify_true = SvTRUE(*svp);
-#endif
+	    unsigned int ssl_mode;
+	    my_bool ssl_enforce = 1;
+	    my_bool ssl_verify = 0;
+	    my_bool ssl_verify_set = 0;
+
+            /* Verify if the hostname we connect to matches the hostname in the certificate */
+	    if ((svp = hv_fetch(hv, "mysql_ssl_verify_server_cert", 28, FALSE)) && *svp) {
+  #if defined(HAVE_SSL_VERIFY) || defined(HAVE_SSL_MODE)
+	      ssl_verify = SvTRUE(*svp);
+	      ssl_verify_set = 1;
+  #else
+	      set_ssl_error(sock, "mysql_ssl_verify_server_cert=1 is not supported");
+	      return NULL;
+  #endif
+	    }
+        if ((svp = hv_fetch(hv, "mysql_ssl_optional", 18, FALSE)) && *svp)
```

Comment:
This throws a compiler warning with GCC because if misleading indentation 

```
dbdimp.c:1937:9: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
         if ((svp = hv_fetch(hv, "mysql_ssl_optional", 18, FALSE)) && *svp)
```

Reply:
thanks - I'll look at that.

Reply:
gah, there si some 4 space indentation, some 2. Not sure what we all agreed upon. MySQL source when I worked there was always 2. Thoughts?

Reply:
also, this doesn't look this way in my editor. I need to figure out what .vimrc to get me what I want.